src/main/scala/firrtl/graph/DiGraph.scala

```diff
@@ -29,12 +28,18 @@ object DiGraph {
         edgeDataCopy(k) += n
       }
     }
-    new DiGraph(edgeDataCopy)
+    new DiGraph[T] {
+      val edges = edgeDataCopy
+    }
   }
 }
 
-/** Represents common behavior of all directed graphs */
-class DiGraph[T] private[graph] (private[graph] val edges: LinkedHashMap[T, LinkedHashSet[T]]) {
+abstract class DiGraph[T] {
```

Comment:
What is the incentive for making this abstract?

Reply:
reverted.